hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/FSHLog.java

```diff
@@ -577,10 +577,9 @@ public void run() {
           //TraceScope scope = Trace.continueSpan(takeSyncFuture.getSpan());
           long start = System.nanoTime();
           Throwable lastException = null;
-          boolean wasRollRequested = false;
           try {
             TraceUtil.addTimelineAnnotation("syncing writer");
-            writer.sync(useHsync);
+            writer.sync(takeSyncFuture.isForceSync());
```

Comment:
Is this right? If useHsync is set, doesn't that mean sync every edit?

Reply:
This was also confusing me a bit but after investigating, I think the logic is correct. As this config can be changed at table level or even mutation level by design, we should follow what we have in the SyncFuture, as it is set when syncing the WAL in HRegion.

Reply:
So the desion allow different table/mutation use different durability, but these table share one WAL writer?

Reply:
Since you can change the durability by mutation, you can not use different WALs for different mutations for the same region? But anyway, you can implement a multi wal strategy to use different WALs for different durability levels.

Reply:
Ok. I went back over this and this is correct as the SyncFuture uses whatever the config is set to when created.

bq. So the desion allow different table/mutation use different durability, but these table share one WAL writer?

Yes.. this could make for some 'interesting' results. IIRC, looking at this w/ Lars a while back, if async WAL and you were trying to manage the syncs from the application, a client could get an ack though all puts hadn't yet syncd'.